addon-test-support/@ember/test-helpers/teardown-context.ts

```diff
@@ -31,13 +40,21 @@ export default function teardownContext(context: TestContext): Promise<void> {
 
       unsetContext();
 
-      return settled();
+      if (waitForSettled) {
+        return settled();
+      }
+
+      return nextTickPromise();
     })
     .finally(() => {
       let contextGuid = guidFor(context);
 
       runDestroyablesFor(CLEANUP, contextGuid);
```

Comment:
can we run this in a potentially unsettled state?

Reply:
Yes, the cleanup itself is unrelated to settledness (we call `owner.destroy()` synchronously just above).